src/main/types/SIOP.types.ts

```diff
@@ -90,15 +90,67 @@ export interface AuthenticationResponsePayload extends JWTPayload {
   did: string;
   registration?: DiscoveryMetadataPayload;
   registration_uri?: string;
-  vp_token?: VerifiablePresentationWrapper[];
+  verifiable_presentations?: VerifiablePresentationPayload[];
+  // fixme All of the above is the id token. We need to create a new interface of the above and reference that here as id_token
+  vp_token?: VerifiablePresentationPayload;
   claims?: ResponseClaims;
 }
+/*
+
+export interface OidcClaimJson {
+  essential?: boolean;
+  value?: string;
+  values?: string[];
+}
+
+export interface OidcClaimRequest {
+  [x: string]: null | OidcClaimJson;
+}*/
+
+export interface IdTokenClaimPayload {
+  verifiable_presentations?: PresentationDefinition[];
+  [x: string]: unknown;
+}
+export interface VpTokenClaimPayload {
+  verifiable_presentation?: PresentationDefinition;
+  [x: string]: unknown;
+}
+
+export interface ClaimOpts {
```

Comment:
based on [embedded-verifiable-present](https://openid.net/specs/openid-connect-4-verifiable-presentations-1_0.html#name-embedded-verifiable-present) and [name-vp-token](https://openid.net/specs/openid-connect-4-verifiable-presentations-1_0.html#name-vp-token) this can't be valid

Reply:
These are options the user of the library provides. These are not payloads. It gets translated into a payload later.

I mentioned the other day to you that you should not pass in payloads as a user because it exposes too many the internals and you loose strength (like in this case having a single object that can end up in multiple forms in the eventual payload)